DIPs/DIP1000.md

```diff
@@ -87,29 +88,72 @@ the function.
 
 #### Reachability vs. lifetime
 
-For each value `v` within a program we define the notion of <i>reachability</i>
+##### Reachability
+
+For each value `v` within a program we define the notion of _reachability_
 denoted as `reachability(v)`, akin to the lexical extent through which the value
 can be accessed.
 
-- For an rvalue, the reachability is the expression within which it is used.
+- For an unnamed temporary (either lvalue such as `array[f()]` or rvalue such as
+`42 + f()`), the reachability is the largest grammatical expression within which
+it is used.
 - For a named variable in a scope, the reachability is the lexical scope of the variable per the language rules.
-- For a module-level variable, reachability is considered infinite. Notation: `reachability(v)` `=` `∞`.
+- For a module-level variable, reachability is considered infinite. Notation:
+`reachability(v)` `=` `∞`.
+
+For values with finite reachability, we denote their reachability as a tuple
+`M:B:E`, where:
+
+- `M` is a string that uniquely identifies the module within the program
+- `B` is an integral that is the index of the character at which reachability
+begins, as follows:
+  - For temporaries, `B` is the index of the first character of the largest
+expression in source code, and `E` is the index of the last character of the
+same largest expression
+  - For named values, `B` is the index of the `;` or `,` that ends the
+definition of the value, and `E` is the index of the `}` that closes the scope
+of the definition.
+  - For `ref` and `out` parameters, `B` is the index of the opening brace `{`
```

Comment:
Although with shorh-hand lambdas we have only a single expression, so may be you can just mention that in the case of lambdas, the parameters are reachable in the whole lambda expression.